### ISSUE TITLE
Can native tx and gorm tx integrate with each other?

### ISSUE THREAD

**Issue by redochenzhen** [go-gorm/gorm#6562]
## Your Question

I have a component using native sql transaction(*sql.DB.BeginTx). Is it possible to integrate with gorm's Transaction? That is to say, can my component and gorm work in the same transaction?

## Expected answer

1. A method to get interanl native *sql.Tx instance in gorm transaction
2. A solution to integrate native tx with gorm tx

**Comment by qqxhb** [comment]
Yes, they can work in the same transaction, but only if both sides use the same underlying transaction object/connection.

Important notes:
- GORM’s `ConnPool` is an interface and is not guaranteed to be `*sql.Tx` (depends on dialector/driver), so there is no universal API to “get the internal *sql.Tx”.

Practical options:

1) Create a native `*sql.Tx` first, then run GORM on top of that tx (recommended / most reliable):

```go
sqlTx, err := sqlDB.BeginTx(ctx, nil)
if err != nil { return err }

defer sqlTx.Rollback()

// Use the same *sql.Tx as ConnPool for GORM
ormDB, err := gorm.Open(dialector, &gorm.Config{ConnPool: sqlTx})
if err != nil { return err }

// Use ormDB for GORM operations, and sqlTx for native SQL operations
// ...

return sqlTx.Commit()
```

2) If your dialector uses `database/sql`, inside `db.Transaction` you can try to type-assert the ConnPool to `*sql.Tx`:

```go
db.Transaction(func(tx *gorm.DB) error {
    if sqlTx, ok := tx.Statement.ConnPool.(*sql.Tx); ok {
        // run native SQL using sqlTx
    }
    // run GORM ops using tx
    return nil
})
```

So: integration is possible, but GORM doesn’t guarantee access to a `*sql.Tx` unless the underlying driver is `database/sql`-based. The safest pattern is to start the native tx and inject it into GORM.

**Comment by redochenzhen** [comment]
it solves my problems. thx!
